core/server/apps/amp/index.js

```diff
@@ -1,15 +1,26 @@
-var router           = require('./lib/router'),
+var router              = require('./lib/router'),
     registerAmpHelpers  = require('./lib/helpers'),
+    config              = require('../../config'),
+    ampIsEnabled;
 
-    // Dirty requires
-    config     = require('../../config');
+ampIsEnabled = function ampIsEnabled() {
+    var ampSettings = config.theme.amp;
+
+    return ampSettings === 'true' ? true : false;
+};
 
 module.exports = {
     activate: function activate(ghost) {
         registerAmpHelpers(ghost);
     },
 
     setupRoutes: function setupRoutes(blogRouter) {
-        blogRouter.use('*/' + config.routeKeywords.amp + '/', router);
+        blogRouter.use('*/' + config.routeKeywords.amp + '/', function ampEnabled(req, res, next) {
```

Comment:
If amp is disabled and my blog runs on a subdirectory, then the redirect does not work.

http://localhost:2368/blog/welcome-to-ghost/amp
http://localhost:2368/welcome-to-ghost/

Because the next middleware/controller which is `next` is the `single` post lookup, see https://github.com/TryGhost/Ghost/blob/lts/core/server/controllers/frontend/index.js#L60

And this is a little tricky, let me try to explain what happens:

1. for a normal post lookup all works as expected
2. but in the case of amp (/blog/welcome-ghost/amp) this https://github.com/TryGhost/Ghost/blob/lts/core/server/controllers/frontend/index.js#L75 is false, because req.path (/welcome-ghost/amp) !== post.url (/welcome-ghost)

`if (post.url !== req.path) {` is actually a correct condition. We compare the generated url from the model with the requested url.

The correct way of solving such a routing problem is using express routers, but this use case is hard, because amp routes match the *end* of a requested url.

So i suggest to redirect the client request to a temporary redirect. The max-age should be low in case amp get's enabled again.

```
// add a reasoning here
res.redirect('/to-the-blog-post-without-amp')
```